### PR TITLE
use external hostname in email subject

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

InetAddress.getLocalHost().getHostName() just get the internal container network FQDN name.
This is fix to `uyuni-server.mgr.internal ` and does not help to identify the origin of the error if you run multiple instances.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Port(s): https://github.com/SUSE/spacewalk/pull/25405

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: java/code/src/com/redhat/rhn/frontend/events/TraceBackAction.java [hunk 1]
```diff
@@ -37,31 +36,21 @@ public class TraceBackAction extends BaseMailAction implements MessageAction {
...
-        }
-        subject.append(" (");
-        subject.append(LocalizationService.getInstance().formatDate(new Date(),
-                evtIn.getUserLocale()));
-        subject.append(")");
-        return subject.toString();
+        return LocalizationService.getInstance().
+                getMessage("web traceback subject", evtIn.getUserLocale()) +
+                // Not sure if getting the local hostname is the correct thing to do
+                // here.  But the traceback emails that I've received seem to do this
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/frontend/events/TraceBackAction.java:42]:
```suggestion
```

This is not true anymore, so should be removed.

File: java/code/src/com/redhat/rhn/taskomatic/task/TaskHelper.java [hunk 2]
```diff
@@ -80,24 +73,17 @@ public static void sendErrorEmail(String messageBody) {
...
      */
     public static void sendTaskoEmail(Integer orgId, String messageBody) {
-        Config c = Config.get();
         LocalizationService ls = LocalizationService.getInstance();
         String[] recipients = MailHelper.getAdminRecipientsFromConfig();
         if (orgId != null) {
             List<String> emails = getActiveOrgAdminEmails(orgId);
             recipients = !emails.isEmpty() ?
-                    emails.toArray(new String[emails.size()]) :
+                    emails.toArray(new String[0]) :
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/taskomatic/task/TaskHelper.java:81]: I don't get why this change?

**Review by aaannz — COMMENTED** [review]
One question and nitpick, but otherwise LGTM

File: java/code/src/com/redhat/rhn/taskomatic/task/TaskHelper.java (hunk 2 above)
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/taskomatic/task/TaskHelper.java:81]: Was a sonarcloud suggestion. toArray takes care that all the elements are returned. The parameter should only give the type. It may make difference when you want to return an array which is bigger than the number of elements, but same size should also work this way.

**Review by aaannz — APPROVED** [review]

**Review by mackdk — APPROVED** [review]
